lib/Dialect/TTNN/Transforms/Passes.cpp

```diff
@@ -98,6 +111,66 @@ class TTNNDeallocate : public impl::TTNNDeallocateBase<TTNNDeallocate> {
   }
 };
 
+class TTNNValidateGraphCapture
+    : public impl::TTNNValidateGraphCaptureBase<TTNNValidateGraphCapture> {
+
+public:
+  using impl::TTNNValidateGraphCaptureBase<
+      TTNNValidateGraphCapture>::TTNNValidateGraphCaptureBase;
+
+  void runOnOperation() final {
+    const std::filesystem::path tmpDirPath =
+        std::filesystem::temp_directory_path();
+
+    const std::string mlirFilePath = tmpDirPath / "module.mlir";
+    const std::string flatBufferFilePath = tmpDirPath / "module.ttnn";
+    const std::string outReportPath = tmpDirPath / "module_graph_capture.json";
+
+    outputTTNNIRFile(mlirFilePath);
+    outputFlatBufferFile(mlirFilePath, flatBufferFilePath);
```

Comment:
How are failures in each stage handled, how do you know you are not reading artefact of some previous compile session?